### ISSUE TITLE
DistributedSampler may sample the same batch every epoch

### ISSUE THREAD

**Issue by yuxinyuan** [NVIDIA/tacotron2#241]
https://github.com/NVIDIA/tacotron2/blob/131c1465b48be60cb5d3b8ab79cfc663e5c47b6a/train.py#L49

It seems that pytorch DistributedSampler does a deterministic shuffle based on epoch. 

I guess this would mean if you don't manually update the sampler's internal state (self.epoch), the dataloader will get the same batch (because of the same random seed) at every epoch in distributed training setup.

I suppose this should be a bug.

**Comment by maxbaluev** [comment]
We think because of this distributed learning doesn't work. And we have to train the model on one video card.
